test/systemtests/network_test.go

```diff
@@ -371,6 +435,72 @@ func (s *systemtestSuite) testNetworkAddDeleteTenant(c *C, encap, fwdmode string
 
 }
 
+func (s *systemtestSuite) testNetworkAddDeleteTenantWithDns(c *C, encap, fwdmode string) {
+
+	var err error
+
+	for i := 0; i < s.basicInfo.Iterations; i++ {
+		var (
+			tenantNames = map[string][]string{}
+			netNames    = []string{}
+			pktTag      = 0
+		)
+
+		for tenantNum := 0; tenantNum < 3; tenantNum++ {
+			tenantName := fmt.Sprintf("tenant%d", tenantNum)
+			c.Assert(s.cli.TenantPost(&client.Tenant{TenantName: tenantName}), IsNil)
+			tenantNames[tenantName] = []string{}
+
+			for networkNum := 0; networkNum < 3; networkNum++ {
+				var v6subnet, v6gateway string
+				if networkNum%2 == 0 {
+					v6subnet = ""
+					v6gateway = ""
+				} else {
+					v6subnet = fmt.Sprintf("1001:%d:%d::/120", tenantNum, networkNum)
+					v6gateway = fmt.Sprintf("1001:%d:%d::254", tenantNum, networkNum)
+				}
+				network := &client.Network{
+					TenantName:  tenantName,
+					NetworkName: fmt.Sprintf("net%d-%d", networkNum, i),
+					Subnet:      fmt.Sprintf("10.%d.%d.0/24", tenantNum, networkNum),
+					Gateway:     fmt.Sprintf("10.%d.%d.254", tenantNum, networkNum),
+					Ipv6Subnet:  v6subnet,
+					Ipv6Gateway: v6gateway,
+					PktTag:      pktTag + 1000,
+					Encap:       encap,
+				}
+
+				logrus.Infof("Creating network %s on tenant %s", network.NetworkName, network.TenantName)
+
+				c.Assert(s.cli.NetworkPost(network), IsNil)
+				netNames = append(netNames, network.NetworkName)
+				tenantNames[tenantName] = append(tenantNames[tenantName], network.NetworkName)
+				pktTag++
+			}
+		}
+
+		// Wait 10 seconds for the call backs to come through
+
+		time.Sleep(10 * time.Second)
```

Comment:
The same pattern is used throughout the system tests. 10 seconds is an extremely generous time given the responses typically come back in well under 1 second. If we really want to have alternate logic, we will need to look at all the system tests since we have the same pattern in all of them. Hence skipping this suggestion for this change/ fix.